File: PMotion_simple_1.cpp
#include<iostream>
#include<cmath>
#include<fstream>

const double g = 9.81;  // Acceleration due to gravity (m/s^2)

double deg2rad(double degrees) { //converting degree input to radian for calculation
    return degrees * M_PI / 180.0;
}

int main() {
    double v0, theta0;
    int steps;

    std::cout << "Enter initial velocity (m/s): ";
    std::cin >> v0;
    std::cout << "Enter launch angle (degrees): "; //could take radian but nobody measures in radian
    std::cin >> theta0;
    std::cout << "Enter number of steps: ";
    std::cin >> steps;

    double v_x0 = v0 * cos(deg2rad(theta0));  // Initial horizontal velocity
    double v_y0 = v0 * sin(deg2rad(theta0));  // Initial vertical velocity
    double flight_time = (2 * v_y0) / g;  // Total flight time
    double time_step = flight_time / steps;  // Time increment for each step

    std::ofstream output_file("Pmotion_output.txt");

    double x = 0.0;  // Initial horizontal position
    double y = 0.0;  // Initial vertical position
    double v_y = v_y0;  // Current vertical velocity
    double y_max = y; //maximum height at initial position
    double t = 0.0; //time variable

    for(int i = 0; i <= steps; i++) {
        if(y < 0) {
            break;
        }

        x = x + v_x0 * time_step;
        y = y + v_y * time_step - 0.5 * g * time_step * time_step;
        v_y = v_y - g*time_step;
        t += time_step;

        output_file << t << " " << x << " " << y << " " << v_x0 << " " << v_y << "\n";
        if(y > y_max) { //update maximum height if current height is greater
            y_max = y;
        }
    }
    
output_file.close();

std::cout << "Maximum height reached: " << y_max << std::endl;
std::cout << "Horizontal distance traveled: " << x << std::endl;

return 0;
}

File: PMotion_simple_1.py
import math

g = 9.81  # Acceleration due to gravity (m/s^2)

v0 = float(input("Enter initial velocity (m/s): "))
theta0 = float(input("Enter launch angle (degrees): "))

steps = int(input("Enter number of steps: "))

v_x0 = v0*math.cos(math.radians(theta0)) # Initial horizontal velocity
v_y0 = v0*math.sin(math.radians(theta0)) # Initial vertical velocity
flight_time = (2*v_y0)/g # Total flight time
time_step = flight_time/steps # Time increment for each step

output_file = open("Pmotion_output.txt", "w")

x = 0.0 # Initial horizontal position
y = 0.0 # Initial vertical position
v_y = v_y0 # Current vertical velocity
y_max = y

t = 0.0
for step in range(steps + 1):
    if(y<0):
            break

    x = x+v_x0*time_step
    y = y+v_y*time_step - 0.5*g*time_step**2
    v_y = v_y - g*time_step
    t = t + time_step

    output_file.write(f"{t:.2f} {x:.2f} {y:.2f} {v_x0:.2f} {v_y:.2f}\n")
    if(y>y_max):
        y_max = y
    
output_file.close()
print(f"Maximum height reached: {y_max:.2f}")
print(f"Horizontal distance traveled: {x:.2f}")